src/target/ext/base/multithread.rs

```diff
@@ -110,6 +111,12 @@ pub trait MultiThreadBase: Target {
     ) -> Option<crate::target::ext::thread_extra_info::ThreadExtraInfoOps<'_, Self>> {
         None
     }
+
+    /// Allow the target to change the active PID from the default fake one.
```

Comment:
This is an interesting approach.

On one hand, my gut reaction was "eeew, this is really the kind of thing we'd want to hide from the user. The whole point of `MultiThreadBase` is that you _shouldn't_ have to care about anything PID related"

...but given the scope of the work required to get `MultiProcessBase` up and running, and how reasonably unintrusive this method is...

I think I'm okay with this. With some tweaks:

I'd like to reframe this method as a "power user" method - i.e: tweak the docs to make it clear that unless you "know what you're doing", this isn't something you should be tweaking. Namely:

- Reword the above-the-fold docs as `Override the reported PID when running in multithread mode (default: 1)`
- Beef up the "below the fold" docs with info as to _why_ this might be useful (i.e: support extended mode attach / switch between processes when using `ExtendedMode::attach`)